pilz_teleoperation/src/pilz_teleoperation/ros_message_serializer.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2020 Pilz GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import genpy
+import rospy
+from collections import OrderedDict
+
+
+class RosMessageSerializer(object):
+    """
+    Serialization and De-Serialization of ROS messages for convenient file printing
+    """
+
+    def __init__(self):
+        self._module_imports = {}
+
+    def write_messages_to_file(self, ros_messages, filename):
+        """
+        Writes a file that can be used with 'import' to restore
+        :param ros_messages: iterable [ordered] dictionary of messages to be serialized.
+                key is the variable name used for serialization and value a ROS message object
+        :param filename: target filename, overwritten if existent
+        """
+        with open(filename, 'w') as f:
+            serialized_message = ""
+            for varname, msg in ros_messages.items():
+                serialized_message += "{} = {}\n".format(varname, self.convert_ros_message_to_python(msg))
+            f.write("\n".join(
+                "from {} import {}".format(module, typename) for typename, module in self._module_imports.items()))
+            f.write("\n\n")
+            f.write(serialized_message)
+            f.write("\n")
+
+    def convert_ros_message_to_python(self, message, indentation=0):
+        """
+        Serialize a ROS msg into a String Object that can be used with eval() afterwards
+        :rtype: String
+        """
+        if not isinstance(message, (genpy.Message, rospy.rostime.Time, rospy.rostime.Duration)):
+            # end of recursion
+            return self._message_to_string(message)
+
+        field_values = self._message_to_stringdict(message, indentation)
+        result = ',\n'.join(
+            '{}{} = {}'.format(" " * (indentation + 2), key, value) for key, value in field_values.items())
+        self._module_imports[type(message).__name__] = type(message).__module__
```

Comment:
type(message).__module__ results in imports like:
```
from geometry_msgs.msg._PoseStamped import PoseStamped
from geometry_msgs.msg._Point import Point
from geometry_msgs.msg._Pose import Pose
from std_msgs.msg._Header import Header
```

so it is using the private modul names, not the finally imported ones. It works, but you may want to think about resolving that, to prevent warnings when using this file

Reply:
currently okay as it is.